### PR TITLE
Fix #8677

### LINKED ISSUE #8677: Group Plans task approval list needs information about task types, better formatting
When a member of a group plan completes a task that requires approval, the group leader sees the approvals as shown in the screenshot below.

1. It's not possible to tell whether the tasks are Habits, Dailies, To-Dos, or Rewards. In the case below, for the first two tasks, one is a Habit and one a Daily. There needs to be an indication of task type.
1. For Habits, it's also not possible to tell if the member clicked the plus or minus button, which is important to know for correctly approving it. It should state whether plus or minus was clicked.
1. There should also be more distinction for the name of the task. At the moment, it blends in with the user information (the task names for the first two tasks here are "clean your room").
1. Markdown and emoji in the task title should be rendered.

image

Note that there is currently an issue with negative clicks on group plans Habits being treated as positive clicks (https://github.com/HabitRPG/habitica/issues/8676). It's possible that point 2 above can't be fixed until that issue is fixed (e.g., if the approval screen currently can't tell if plus or minus was clicked).

### PR DESCRIPTION
**Pull request by jsjoeio**
[//]: # (https://github.com/HabitRPG/habitica/issues/8677)
Fixes https://github.com/HabitRPG/habitica/issues/8677

### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)
This fix aims to do three things on the Group Plans Task Approval List:

1. Indicate the task type for which the member of the party is requesting approval.
2. Differentiate between the username and the task using quotation marks
3. Render markdown and emojis in the task title

This issue also mentions this bug "For Habits, it's also not possible to tell if the member clicked the plus or minus button, which is important to know for correctly approving it. It should state whether plus or minus was clicked." However, @Alys notes that it may not be possible to solve this bug until #8676 is fixed. So we did not attempt to fix that bug in this pull request.

In order to solve these bugs, we had to modify the groups.json file to add in the task "type." Then we added the type as approval.type on the approvalTitle function on the groupApprovalsController page.

Last, we needed to add the markdown tags on the group-tasks-approvals.jade file so that any markdown or emoji used in the task titles would be rendered correctly on the Group Plans Task Approval List page. 

Here is what the original Group Plans Task Approval List looked like:
7e0787b4-235a-11e7-8669-81b358be06c9

Here is what it looks like with the changes (option 1)
option1

**^Note:** I realize that the Approve button is slightly below where it should be and there's a strange white space added to the task block element. We tried various things to fix it and the only thing we could come up with was this (option 2):
option2
^The code to display it formatted as so(option 2) is included in the file group-tasks-approvals.jade but commented out (see below). I was told in the Aspiring Blacksmiths' Guild to submit a pull request and ask for guidance on how to solve this formatting issue.

[//]: # (b686be6c-6db5-42c5-b4de-92ae66031411)

----
UUID: b686be6c-6db5-42c5-b4de-92ae66031411  - @jjprevite
and UUID: e5026bf1-acca-437f-a5f7-8cc738b74f02  - @razzlen

This issue was the work of both myself and @razzlepdx (her Github username). It was our first time contributing to an opensource project so we wanted to work on it together, thanks to the encouragement from Habitica user @AccioBooks!!

If possible, we would like to credit our party and group- FCW or Future Coding Wizards. Although only @razzlen and myself worked on this issue, we represent a Habitica party who are all learning to code and contribute to opensource projects which include myself, @razzlen, @Nyaemal Hrruna and @seereuben.

### CODE REVIEW

File: website/client/readme.md [hunk 1]
```diff
@@ -1,20 +0,0 @@
-#Running
```
Comment (librarianmage) [inline website/client/readme.md:1]: Why did you delete this file?

File: website/client-old/js/components/groupApprovals/groupApprovalsController.js [hunk 2]
```diff
@@ -9,7 +9,7 @@ habitrpg.controller('GroupApprovalsCtrl', ['$scope', 'Tasks',
     };
 
     $scope.approvalTitle = function (approval) {
-      return env.t('approvalTitle', {text: approval.text, userName: approval.userId.profile.name});
+      return env.t('approvalTitle', {type: approval.type, text: approval.text, userName: approval.userId.profile.name});
```
Comment (librarianmage) [inline website/client-old/js/components/groupApprovals/groupApprovalsController.js:12]: What does this addition do?

File: website/views/options/social/groups/group-tasks-approvals.jade [hunk 3]
```diff
@@ -9,5 +9,19 @@ script(type='text/ng-template', id='partials/groups.tasks.approvals.html')
   .panel-group(ng-repeat="approval in group.approvals")
     .panel.panel-default
       .panel-heading
-        span {{approvalTitle(approval)}}
-        a.btn.btn-sm.btn-success.pull-right(ng-click="approve(approval.group.taskId, approval.userId._id, approval.userId.profile.name, $index)")=env.t('approve')
+            span
```
Comment (librarianmage) [inline website/views/options/social/groups/group-tasks-approvals.jade]: I believe the indentation should be two spaces, not four

File: website/client/readme.md (hunk 1 above)
Comment (jsjoeio) [reply website/client/readme.md:1]: I must have deleted it by accident. How should I go about re-adding it to my branch?

File: website/client-old/js/components/groupApprovals/groupApprovalsController.js (hunk 2 above)
Comment (jsjoeio) [reply website/client-old/js/components/groupApprovals/groupApprovalsController.js:12]: By adding this piece in--type: approval.type-- we are able to access the task type on the groups.json file and eventually display it on the Group Plans Tasks Awaiting Approval List page.

File: website/views/options/social/groups/group-tasks-approvals.jade (hunk 3 above)
Comment (jsjoeio) [reply website/views/options/social/groups/group-tasks-approvals.jade]: Thank you! I'll fix this.

File: website/client/readme.md (hunk 1 above)
Comment (jsjoeio) [reply website/client/readme.md:1]: I'm trying to add the file again but it won't let me because there is already a file with the title README.md that has the same contents. Is it necessary to have both?

**Comment by jsjoeio** [comment]
Hey @MathWhiz - should I make all the changes you suggested and then make a new pull request? I'm realizing that this pull request looks "sloppy" because of my commits. I could definitely clean it up a bit and make it more clear and understandable for everyone at Habitica by making 3 commits rather than 5. What are your thoughts?

**Comment by deltaidea** [comment]
@jjprevite Git has powerful options for changing commit history. Your best option would be *interactive rebase*. Please see [Rewriting History](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) on their website. TL;DR is you have two options:
- Run `git rebase -i <hash>` where hash is the **parent** of the first commit that you want to change. This will walk you through all commits since parent on the current branch.
- Use a GUI like SourceTree that has a visual representation of the interactive rebase.

After you've changed branch history locally, it's now in conflict with the remote version. In this case, that's expected and you know what you're doing, so run `git push --force origin <branch>`.

> we represent a Habitica party who are all learning to code and contribute to opensource projects

You guys are beyond awesome.

**Comment by jsjoeio** [comment]
@deltaidea Thank you for the suggestions! So I think it'd be easier to delete this pull request, delete the branch then redo the changes in commits that will make sense to everyone.

**Comment by jsjoeio** [comment]
I'm going to close this, redo it and submit a new pull request.
